jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java

```diff
@@ -509,15 +509,50 @@ protected void copyRequestHeaders(HttpServletRequest clientRequest, Request prox
 
     protected void addProxyHeaders(HttpServletRequest clientRequest, Request proxyRequest)
     {
-        addViaHeader(proxyRequest);
+        addViaHeader(clientRequest, proxyRequest);
         addXForwardedHeaders(clientRequest, proxyRequest);
     }
 
+    /**
+     * Adds the HTTP Via header to the proxied request.
+     *
+     * @deprecated Use {@link #addViaHeader(HttpServletRequest, Request)} instead.
+     * @param proxyRequest the request being proxied
+     */
+    @Deprecated
     protected void addViaHeader(Request proxyRequest)
```

Comment:
We're calling the new method and the result is the same for 1.1 clients. So, for those that _have not_ overridden this method, we're fixing it for 2.0 and non-HTTP clients. For those that have overridden it, we're doing the same thing in case they call super. 

I don't see any backward compatibility issue here.

Reply:
Never mind. I eventually got what you were saying :)